blog/_posts/2016-10-01-julia-0.5-highlights.md

```diff
@@ -0,0 +1,726 @@
+eventually, if you're allocating, however, the garbage collector does kick in and then the piper gets paid.
+
+The last major missing piece of vectorized call syntax is that in the future dotted operators will cease to be their own independent operators and simply become the vectorized form of the corresponding scalar operator.
+In other words, instead of `.+` being a function as it is now, with it's own potentially unrelated behavior independent of `+`, it sill simply be the vectorized form of `+`.
+Thus, in the next version of Julia, when you write `A .+ B` it will mean `broadcast(+, A, B)`.
+Moreover, this will participate in the same syntax-level fusion as other vectorized calls, so expressions like `exp.(log.(X) .+ log.(Y))` will translate into:
+
+    broadcast((x, y)->exp(log(x) + log(y)), X, Y)
+
+This change will complete the transition of Julia to using generalized vectorized function application with syntax-level loop fusion, which will make the story for writing allocation-free array code much stronger.
+
+## Comprehensions
+
+Julia's array comprehensions have always supported some advanced features such as iterating with several variables to produce multidimensional arrays.
+This release rounds out the functionality of comprehensions with two additional features: nested generation with multiple `for` clauses, and filtering with a trailing `if` clause.
+To demonstrate these features, consider making a dollar (100¢) using quarters (25¢), dimes (10¢), nickels (5¢) and pennies (1¢).
+We can generate an array of tuples of total values in each kind of coin by using a comprehension with nested `for` clauses:
+
+    julia> change = [(q,d,n,p) for q=0:25:100 for d=0:10:100-q for n=0:5:100-q-d for p=100-q-d-n]
+    242-element Array{NTuple{4,Int64},1}:
+     (0,0,0,100)
+     (0,0,5,95)
+     (0,0,10,90)
+     (0,0,15,85)
+     (0,0,20,80)
+     (0,0,25,75)
+     ⋮
+     (75,10,5,10)
+     (75,10,10,5)
+     (75,10,15,0)
+     (75,20,0,5)
+     (75,20,5,0)
+     (100,0,0,0)
+
+There are a few notable differences from the multidimensional array syntax:
+
+- Each iteration is a new `for` clause, rather than a single compound iteration separated by commas.
+- Each successive `for` clause *can* refer to variables from the previous clauses.
+- The result is a single flat vector regardless of how many `for` clauses there are.
+
+The tuple `(q,d,n,p)` in the comprehension body is a breakdown of monetary value into quarters, dimes, nickels and pennies.
+Note that the iteration range for `p` isn't a range at all, it's a single value, `100-q-d-n`, the unique number guaranteeing that each tuple adds up to a dollar.
+[This relies on the fact that a number behaves like an immutable zero-dimensional container, holding only itself, a behavior which is sometimes convenient but which has been the subject of significant debate.
```

Comment:
Also parentheses would be more consistent with your asides in other places in this post